plugins/modules/mount.py

```diff
@@ -835,7 +912,25 @@ def main():
             if changed and not module.check_mode:
                 res, msg = remount(module, args)
                 changed = True
+
+            # When 'state' == 'ephemeral', we don't know what is in fstab, and 'changed' is always False
+            if state == 'ephemeral':
+                # If state == 'ephemeral', check if the mountpoint src == module.params['src']
+                # If it doesn't, fail to prevent unwanted unmount or unwanted mountpoint override
+                if _is_same_mount_src(module, args):
+                    res, msg = remount(module, args)
+                    changed = True
+                else:
+                    module.fail_json(
+                        msg=(
+                            'Ephemeral mount point is already mounted with a different '
+                            'source than the specified one. Failing in order to prevent an '
+                            'unwanted unmount or override operation. Try replacing this command with '
+                            'a "state: unmounted" followed by a "state: ephemeral", or use '
+                            'a different destination path.'))
+
```

Comment:
When *state* is `ephemeral` or `mounted`, it tests if the provided mountpoint has already a volume mounted on it.

In the case of `ephemeral`, there is no *fstab* file to provide an authoritative answer to the question "What should be mounted here". To avoid an unwanted unmount and a possible service outage, the module fails if the volume already mounted on the mountpoint is not the same as the one provided in *src* (this is checked with the new function `_is_same_mount_src`).